pkg/depository/get.go

```diff
@@ -17,17 +17,144 @@ limitations under the License.
+	query := url.Values{}
+	from, _ := cmd.Flags().GetInt("from")
+	if from != 0 {
+		query.Add("from", fmt.Sprintf("%d", from))
+	}
+	size, _ := cmd.Flags().GetInt("size")
+	if size != 0 {
+		query.Add("size", fmt.Sprintf("%d", size))
+	}
+	kid, _ := cmd.Flags().GetString("kid")
+	if kid != "" {
+		query.Add("kid", kid)
+	}
+	name, _ := cmd.Flags().GetString("name")
+	if name != "" {
+		query.Add("name", name)
+	}
+	contentName, _ := cmd.Flags().GetString("contentName")
+	if contentName != "" {
+		query.Add("contentName", contentName)
+	}
+
+	return fmt.Sprintf("%s?%s", common.ListDepository, query.Encode())
+}
+
+var headers = []string{"index", "kid", "platform", "operator", "owner", "blockNumber", "time"}
+
+type Options struct {
+	genericclioptions.IOStreams
+}
+
+func NewGetDepositoryCmd(option Options) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "depository [KID]",
+		Short: "Get one or more depositories",
 		RunE: func(cmd *cobra.Command, args []string) error {
-			fmt.Println("get depositories")
+			// FIXME: the host should be read from the configuration file.
+			host, _ := cmd.Flags().GetString("host")
+			if host == "" {
+				return fmt.Errorf("no host provided")
+			}
+			if len(args) == 0 {
+				u := fmt.Sprintf("%s%s", host, ConstructQuery(cmd))
+				x, err := DoGetReq(u, nil)
+				if err != nil {
+					fmt.Fprintf(option.ErrOut, "Error failed to get depository: %s\n", err.Error())
+					return err
+				}
+				var data struct {
+					Data  []Depository `json:"data"`
+					Count int64        `json:"count"`
+				}
+				if err := json.Unmarshal(x, &data); err != nil {
+					fmt.Fprintf(option.ErrOut, "unmarhsal response error %s\n", err.Error())
+					return err
+				}
+				xx := make([]printer.Printer, len(data.Data))
+				for i := 0; i < len(data.Data); i++ {
+					xx[i] = data.Data[i]
+				}
+				printer.Print(option.Out, headers, xx)
+				return nil
+			}
+			errMsg := make([]string, 0)
+			pobj := make([]printer.Printer, 0)
+			for _, kid := range args {
+				u := fmt.Sprintf("%s%s", host, fmt.Sprintf(common.GetDepository, kid))
+				x, err := DoGetReq(u, nil)
+				if err != nil {
+					errMsg = append(errMsg, err.Error())
+					continue
+				}
+				var o Depository
+				if err := json.Unmarshal(x, &o); err != nil {
+					errMsg = append(errMsg, err.Error())
+					continue
+				}
+				pobj = append(pobj, o)
+			}
+			printer.Print(option.Out, headers, pobj)
+			for _, e := range errMsg {
+				fmt.Fprintln(option.ErrOut, e)
+			}
 			return nil
 		},
 	}
+
+	cmd.Flags().IntP("from", "f", 0, "pagination")
+	cmd.Flags().IntP("size", "s", 10, "pagination size")
+	cmd.Flags().StringP("kid", "k", "", "search depository by kid")
+	cmd.Flags().StringP("name", "n", "", "search depository by name")
+	cmd.Flags().StringP("contentName", "c", "", "search depository by content name")
+	cmd.Flags().StringP("host", "", "", "bc-saas server")
+
+	return cmd
+}
+
+func DoGetReq(u string, heades map[string]string) ([]byte, error) {
```

Comment:
As other blockchain resources also need do `Get`, it's better to put this into a common library or utilities.

BTW... post/delete as well